operator/controllers/manifest_controller.go

```diff
@@ -93,44 +113,27 @@ func (r *ManifestReconciler) HandleInitialState(ctx context.Context, _ *logr.Log
 
 func (r *ManifestReconciler) HandleProcessingState(ctx context.Context, logger *logr.Logger, manifestObj *v1alpha1.Manifest) error {
 	chartCount := len(manifestObj.Spec.Charts)
-	deployJob := make(chan v1alpha1.ChartInfo, chartCount)
-	results := make(chan error, chartCount)
-
-	go func() {
-		endState := v1alpha1.ManifestStateReady
-		var err error
-		for a := 1; a <= chartCount; a++ {
-			err = <-results
-			if err != nil {
-				logger.Error(err, "chart installation failure!!!")
-				endState = v1alpha1.ManifestStateError
-				break
-			}
-		}
 
-		latestManifestObj := &v1alpha1.Manifest{}
-		namespacedName := client.ObjectKey{Name: manifestObj.Name, Namespace: manifestObj.Namespace}
-		if err := r.Get(ctx, namespacedName, latestManifestObj); client.IgnoreNotFound(err) != nil {
-			logger.Error(err, "unexpected error", "resource", namespacedName)
-		}
-
-		if err := r.updateManifestStatus(ctx, latestManifestObj, endState, "manifest charts installed!"); err != nil {
-			logger.Error(err, "error updating status", "resource", namespacedName)
-		}
-	}()
-
-	r.Workers.StartWorkers(deployJob, results, r.HandleCharts)
+	go r.ResponseHandlerFunc(ctx, chartCount, logger, client.ObjectKey{Namespace: manifestObj.Namespace, Name: manifestObj.Name})
 
 	// send job to workers
 	for _, chart := range manifestObj.Spec.Charts {
-		deployJob <- chart
+		r.DeployChan <- DeployInfo{&chart, CreateMode}
 	}
 
-	close(deployJob)
 	return nil
 }
 
-func (r *ManifestReconciler) HandleDeletingState(_ context.Context) error {
+func (r *ManifestReconciler) HandleDeletingState(ctx context.Context, logger *logr.Logger, manifestObj *v1alpha1.Manifest) error {
```

Comment:
This function is the same with `HandleProcessingState`, the only difference is the DeletionMode/CreateMode parameter. Maybe you can extract this implementation into some common, private function?

Reply:
done